src/software/geom/util.cpp

```diff
@@ -1251,6 +1116,57 @@ std::optional<Segment> mergeFullyOverlappingSegments(Segment segment1, Segment s
     }
 }
 
+std::vector<Segment> reduceParallelSegments(std::vector<Segment> segments)
+{
+    if (segments.size() == 0)
+    {
+        return std::vector<Segment>();
+    }
+    else if (segments.size() == 1)
+    {
+        // If there is only 1 segments, it is unique and should be returned
+        return segments;
+    }
```

Comment:
Why do we need these checks? Shouldn't these cases just be handled below?

Reply:
I think this comment might be outdated with the new functions I added

Reply:
You are correct, removed this if-case